stanford_alt_check.forms.inc

```diff
@@ -2,124 +2,532 @@
+    current_path(),
+    array('query' => array($get)),
+  );
+}
+
+/**
+ * Save form submit.
+ */
+function stanford_alt_check_form_submit($form, &$form_state) {
+  $num_changed = 0;
+  foreach ($form_state['images'] as $key => $image_data) {
+    if ($form_state['values'][$key]['alt'] && $form_state['values'][$key]['alt'] != $image_data['Alt Attribute']) {
+      $num_changed++;
+      $entity = entity_load_single($image_data['entity_info']['entity_type'], $image_data['entity_info']['entity_id']);
+      if ($image_data['type'] == 'wysiwyg') {
+        _stanford_alt_check_wysiwyg_set($entity, $form_state['values'][$key]['alt'], $image_data);
+      }
+      else {
+        $field = $image_data['entity_info']['field_name'];
+        $delta = $image_data['delta'];
+        $entity->{$field}[LANGUAGE_NONE][$delta]['alt'] = $form_state['values'][$key]['alt'];
+      }
+      entity_save($image_data['entity_info']['entity_type'], $entity);
+    }
+  }
+  drupal_set_message(t('Alt text updated for %num images.', array('%num' => $num_changed)));
+}
+
+/**
+ * Sets the alt text in the entity using the values from $value.
+ *
+ * @param object $entity
+ *   The main object with the field data.
+ * @param string $alt
+ *   The alt to set.
+ * @param array $image_data
+ *   The submitted data.
+ */
+function _stanford_alt_check_wysiwyg_set($entity, $alt, array $image_data) {
+  $field = $image_data['entity_info']['field_name'];
+  $field_items = field_get_items($image_data['entity_info']['entity_type'], $entity, $field);
+  $field_item = $field_items[$image_data['delta']];
+  $dom = new DOMDocument();
+  libxml_use_internal_errors(TRUE);
+  $dom->loadHTML(check_markup($field_item['value'], $field_item['format']), LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD);
+  /** @var DOMElement $image */
+  foreach ($dom->getElementsByTagName('img') as $position => $image) {
+    if ($position == $image_data['position'] && $image->getAttribute('src') == $image_data['raw_url']) {
+      $image->setAttribute('alt', $alt);
+      break;
+    }
+  }
+  $field_item['value'] = $dom->saveHTML();
+  $entity->{$field}[LANGUAGE_NONE][$image_data['delta']] = $field_item;
+}
+
+/**
+ * Build the rows for the table from the data for text fields.
+ *
+ * @param array $rows
+ *   Any existing rows to add to.
+ * @param string|null $filter
+ *   If the data should be filtered.
+ *   Possible values:
+ *     'filled': images with populated alts.
+ *     'empty': images with empty alts.
+ */
+function _stanford_alt_check_wysiwgy_rows(&$rows = array(), $filter = NULL) {
+  foreach (_stanford_alt_check_wysiwgy_fields() as $field) {
+    $field_query = new EntityFieldQuery();
+    $field_query->fieldCondition($field['field_name'], 'value', '%<img%', 'like');
+    $results = $field_query->execute();
+    foreach ($results as $entity_type => $entities) {
+      foreach (array_keys($entities) as $entity_id) {
+        $entity = entity_load_single($entity_type, $entity_id);
```

Comment:
Instead of loading a single entity at a time use the entity_load_multiple option and iterate over the results of that array. This will save on the number of mysql queries made to the DB.